macos/library/FLEChannels.h

```diff
@@ -0,0 +1,108 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <Foundation/Foundation.h>
+
+/**
+ * An object encapsulating a method call from Flutter.
+ *
+ * TODO: Move serialization details into a method codec class, to match mobile Flutter plugin APIs.
+ */
+@interface FLEMethodCall : NSObject
+
+/**
+ * Returns a new FLEMethodCall created from a JSON message received from the Flutter engine.
+ */
++ (nullable instancetype)methodCallFromMessage:(nonnull NSDictionary *)message;
+
+/**
+ * Initializes an FLEMethodCall. If |arguments| is provided, it must be serializable to JSON.
+ */
+- (nonnull instancetype)initWithMethodName:(nonnull NSString *)name
+                                 arguments:(nullable id)arguments NS_DESIGNATED_INITIALIZER;
+- (nonnull instancetype)init NS_UNAVAILABLE;
+
+/**
+ * The name of the method being called.
+ */
+@property(readonly, nonatomic, nonnull) NSString *methodName;
+
+/**
+ * The arguments to the method being called, if any.
+ *
+ * This object must be serializable to JSON.
```

Comment:
You might want to note somewhere that NSJSONSerialization is what does the transformation.

Reply:
I added a link to https://docs.flutter.io/flutter/services/JSONMessageCodec-class.html since that explicitly lists the supported types (and the goal is to move to having an FLEJSONMessageCodec, as well as versions of the others).